api/templates/design.go

```diff
@@ -767,15 +772,54 @@ var Template = Type("Template", func() {
 		MaxLength(20)
 		Example("v1.2.3")
 	})
+	Required("id", "name", "deprecated", "author", "description", "defaultVersion")
+})
+
+var Template = Type("Template", func() {
+	Reference(TemplateBase)
+	Description("Template.")
+	Attribute("id")
+	Attribute("name")
+	Attribute("deprecated")
+	Attribute("author")
+	Attribute("description")
+	Attribute("defaultVersion")
+	Attribute("categories", ArrayOf(String), "List of categories the template belongs to.", func() {
+		Example([]string{"E-commerce", "Other", "Social Media"})
```

Comment:
Okrem toho, som zoradil fields aby sa response lepsie citala/debugovala.

Predtym tu bolo `Extend`, ktory rozsiri typ, ale nevyhoda je, ze vsetky parent fields da na koniec.

Tak som to nahradil za `Reference`, kde musis specifikovat field znova, ale uz iba field name, bez porobnosti.
Takto je to potom logicky zoradene (vid E2E tests).